scripts/categorizer.py

```diff
@@ -6,12 +6,81 @@ def find_empty_lines(line):
     :param line:
     :return:
     """
+
     if line.strip("\t").strip(" ").strip("\n") == '':
         return 1
 
     return 0
 
+def find_comment_lines(lines):
+    """
+    Tries to categorize comment lines.
+    We treat comment lines as non-code or something that can
+    cause false positives for other categorizers.
+
+    Also comments + multi line is a double categorization.
+    We are trying to keep categorizations mostly separate from each other,
+    but this one breaks that rule.
+
+    :param line:
+    :return:
+    """
+
+    comment_lines = []
+    multiline_comment = False
+    for line in lines:
+
+        # Checks for a regular comment
+        if line.strip(" ").strip("\t")[0] == "#":
+             comment_lines.append(1)
+             continue
+        # Checking for multiline comment start or end
+        elif ("\"\"\"" in line or "'''" in line) \
+            and "if" not in line:
+            # [ "if" not in line ^ ] is a hack that can come back to hurt me.
+
+
+            comment_lines.append(1)
+
+            # Closes or Opens a multi line comment block
+            if multiline_comment:
+                multiline_comment = False
+            else:
+                multiline_comment = True
+
+            continue
 
+        # Part of a multi line comment
+        if multiline_comment:
+            comment_lines.append(1)
+            continue
+
+        # If it hits nothing else then it appends a zero
```

Comment:
So this is the default case if this was a switch case statement.

Reply:
Maybe add comments which show that parallelism -> default case for this one, multi-line comment case for the one above, and regular comment case to the first one.

Reply:
I need to think this one over...